core/src/main/scripts/importer/validateData.py

```diff
@@ -3229,87 +3263,228 @@ def parse_q_value(self, value, column_number):
+
+    @classmethod
+    def _set_prior_validated_header(cls, header_names):
+        GsvaWiseFileValidator.prior_validated_header = header_names
+    
+    @classmethod
+    def _get_prior_validated_feature_ids(cls):
+        return GsvaWiseFileValidator.prior_validated_feature_ids
+
+    @classmethod
+    def _set_prior_validated_feature_ids(cls, feature_ids):
+        GsvaWiseFileValidator.prior_validated_feature_ids = feature_ids
+
+    @classmethod
+    def _get_prior_validated_sample_ids(cls):
+        return GsvaWiseFileValidator.prior_validated_sample_ids
+
+    @classmethod
+    def _set_prior_validated_sample_ids(cls, sample_ids):
+        GsvaWiseFileValidator.prior_validated_sample_ids = sample_ids
+
+    @classmethod
+    def _get_message_features_do_not_match(cls):
+        return "Gene sets column in score and p-value file are not equal."
+
+
+
+class TreatmentWiseFileValidator(MultipleDataFileValidator, metaclass=ABCMeta):
+    """Groups multiple treatment response files from a study to ebsure consistency.
+
+    All Validator classes that check validity of different treatment response data
+    types in a study should inherit from this class.
+    """
+    prior_validated_sample_ids = None
+    prior_validated_feature_ids = None
+    prior_validated_header = None
+    REQUIRED_HEADERS = ['treatment_id']
+    OPTIONAL_HEADERS = ['name', 'description', 'url']
+    UNIQUE_COLUMNS = ['treatment_id','name']
+
+    def __init__(self, *args, **kwargs):
+        super(TreatmentWiseFileValidator, self).__init__(*args, **kwargs)
+
+    def parseFeatureColumns(self, nonsample_col_vals):
+        self.checkDifferentNameInDb(nonsample_col_vals)
+        return MultipleDataFileValidator.parseFeatureColumns(self, nonsample_col_vals)
+    
+    def checkDifferentNameInDb(self, nonsample_col_vals):
+        # Check for different combinations of treatment_id and treatment_name
+        # in the database. If true, raise warnings for each discrepancy.
+        nonsample_cols =  self.nonsample_cols
+        if (nonsample_cols.index("name") != None):
+
+            treatmentId = nonsample_col_vals[ nonsample_cols.index("treatment_id") ]
+            fileTreatmentName = nonsample_col_vals[ nonsample_cols.index("name") ]
+            dbTreatmentName = self.portal.treatment_map[treatmentId]["name"]
+
+            if (dbTreatmentName != fileTreatmentName):
+                    self.logger.warning(
+                    "Name " + fileTreatmentName + " for treatment `" + treatmentId
+                    + "` is different from name `"+ dbTreatmentName + "` present in the cBioPortal database. "
+                    + "Treatment names in cBioPortal always reflect treatment names in the last imported study. ",
+                    extra={'line_number': self.line_number,
+                            'cause': fileTreatmentName})
+
+    @classmethod
+    def _get_prior_validated_header(cls):
+        return TreatmentWiseFileValidator.prior_validated_header
```

Comment:
Changed for all occurences.

Reply:
As I explained in [another comment](https://github.com/cBioPortal/cbioportal/pull/5460#discussion_r257662107), I suggest just going with `@staticmethod` for now, to remove the unused `cls` argument.

Reply:
Also, please just push the commit if you think it's ready to notify reviewers about.

Reply:
I cannot get this to work in any other way. I have reverted back to class name references.

Reply:
I agree; I was wrong there. You wrote these classes specifically to store state on their own class level, so they can't be more flexible. Static methods that reference that specific class level, instead of class methods, are the way to go.